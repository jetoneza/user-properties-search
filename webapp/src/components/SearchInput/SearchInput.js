import React from 'react';
import { Input, Form, Grid } from 'semantic-ui-react';

export default class SearchInput extends React.PureComponent {
  state = {
    value: '',
  }

  handleSubmit = (e) => {
    e.preventDefault();

    this.props.onSubmit(this.state.value);
  }

  handleChange = (e) => {
    this.setState({ value: e.target.value });
  }

  render() {
    const { value } = this.state;

    return (
      <Grid centered>
        <Grid.Column width={16}>
          <Form onSubmit={this.handleSubmit}>
            <Input
              value={value}
              onChange={this.handleChange}
              action={{ icon: 'search' }}
              placeholder="Search" />
          </Form>
        </Grid.Column>
      </Grid>
    );
  }
}
