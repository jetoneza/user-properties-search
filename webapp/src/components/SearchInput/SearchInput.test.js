import React from 'react';
import renderer from 'react-test-renderer';
import sinon from 'sinon';

import { mount } from 'enzyme';

import SearchInput from './SearchInput';

describe('<SearchInput /> test', () => {
  const onSubmit = jest.fn();

  test('Should update state on input change', () => {
    const wrapper = mount(<SearchInput onSubmit={onSubmit} />);

    const input = wrapper.find('input').first();

    input.simulate('change', { target: { value: 'John Doe' } });

    expect(wrapper.state().value).toEqual('John Doe');
  });

  test('Should call onSubmit prop on handleSubmit', () => {
    const onSubmitCallback = sinon.spy();
    const wrapper = mount(<SearchInput onSubmit={onSubmitCallback} />);

    const form = wrapper.find('form');
    form.simulate('submit');

    expect(onSubmitCallback.called).toBeTruthy();
  });

  test('Should render input', () => {
    const component = <SearchInput onSubmit={onSubmit} />;
    const wrapper = mount(component);

    expect(wrapper.find('form').length).toEqual(1);
    expect(wrapper.find('input').length).toEqual(1);

    const renderedComponent = renderer.create(component);
    const tree = renderedComponent.toJSON();

    expect(tree).toMatchSnapshot();
  });
});


