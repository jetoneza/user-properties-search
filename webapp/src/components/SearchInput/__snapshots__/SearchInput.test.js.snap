// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> test Should render input 1`] = `
<div
  className="ui centered grid"
>
  <div
    className="sixteen wide column"
  >
    <form
      action={undefined}
      className="ui form"
      onSubmit={[Function]}
    >
      <div
        className="ui action input"
      >
        <input
          disabled={undefined}
          onChange={[Function]}
          placeholder="Search"
          tabIndex={undefined}
          type="text"
          value=""
        />
        <button
          className="ui icon button"
          disabled={undefined}
          onClick={[Function]}
          role="button"
          tabIndex={undefined}
        >
          <i
            aria-hidden="true"
            className="search icon"
          />
        </button>
      </div>
    </form>
  </div>
</div>
`;
